Userland/Utilities/man.cpp

```diff
@@ -22,9 +21,14 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
-static ErrorOr<pid_t> pipe_to_pager(DeprecatedString const& command)
+static ErrorOr<pid_t> pipe_to_pager(StringView command)
 {
-    char const* argv[] = { "Shell", "-c", command.characters(), nullptr };
+    StringBuilder builder;
+    TRY(builder.try_append(command));
+    TRY(builder.try_append('\0'));
+    auto shell_command = builder.string_view().characters_without_null_termination();
```

Comment:
Is this really the least bad way to get a null-terminated buffer now? (Not saying that it's not, just wondering)

Reply:
Also I don't love it at all that we're spawning the Shell here instead of just executing the command directly. For one thing this adds (another layer of) unexpected shell-escaping to everything. This has nothing to do with this PR though, so let's not block it on that.

Reply:
> Is this really the least bad way to get a null-terminated buffer now? (Not saying that it's not, just wondering)

I'm really not too sure, this was what I had seen suggested in the SerenityOS Discord so I just went with it. If there is a better approach then I am all ears to it.

>  I don't love it at all that we're spawning the Shell here instead of just executing the command directly

I did wonder about that myself, but I decided to put that to one side and press on with removing the `DeprecatedString`; mostly to avoid getting lost down a rabbit hole.

Reply:
I have just been thinking a bit more on the idea of not calling out to the shell for the pager, and I'm beginning to wonder how that would work?

The list of pager applications is not fixed and I'm not sure how you would start any arbitrary command at runtime without using the shell. Unless of course the list of pagers _does_ become finite, but that seems like a downgrade to me.

Reply:
I might be missing something obvious, but why would you need a shell to execute a command, can't you just `execve` (or `posix_spawn`) it directly? If you really need the shell for something, you can always specify `Shell -c whatever` as the command.

Reply:
Ah, so instead of doing `posix_spawnp(Shell -c `shell_command`,` just do `posix_spawnp(command)`?

Reply:
Yep, that's what I mean.

But again, this could be done separately from the String changes.

Reply:
Ah okay, that makes sense.

> But again, this could be done separately from the String changes.

Agreed, I would prefer to do that in a separate PR.

Reply:
> Is this really the least bad way to get a null-terminated buffer now?

For the task of "get rid of DeprecatedString" it is. In the future, we likely want to look into a better abstraction for that (searching Discord for `CString` should bring up the latest discussion).

Previously, we would just copy the StringView into a DeprecatedString (which guarantees null termination), but then we just get back a usage of DeprecatedString, which is counterproductive.